uiowa/uiowa
# Feature: uiowa_core module

# Site Information form updates
Resolves #1273 
In addition to adding fields to the Site Information form, this also changes the way that uids_base theme choose which version of the header to display. First it checks to see if the site path matches 'sites/uiowa.edu' and if so uses the homepage header. Second it checks to see if the `has_parent` value is true and if so, it uses the parent org name and url to show the "tertiary" header. If neither of the previous two conditions are met, then it uses the "secondary" header.

## To test
1. `blt ds --site=uiowa.edu`
2. Check that https://uiowa.local.drupal.uiowa.edu/ is still showing the correct header.
3. `blt ds --site=immuno.grad.uiowa.edu`
4. `drush @gradimmuno.local uli` and follow link.
5. Visit http://gradimmuno.local.drupal.uiowa.edu/admin/appearance/settings/uids_base and save with the default settings.
6. Check that https://gradimmuno.local.drupal.uiowa.edu/ is showing the "inline" header:
    <img width="1366" alt="Screen Shot 2020-04-10 at 6 12 58 PM" src="https://user-images.githubusercontent.com/709009/79028779-0c7b4d80-7b57-11ea-8433-13f4afab553e.png">
7. Visit http://gradimmuno.local.drupal.uiowa.edu/admin/appearance/settings/uids_base and change the Site name display to "Display below the IOWA bar" and save.
8. Check that https://gradimmuno.local.drupal.uiowa.edu/ is showing the "below" header:
    <img width="1511" alt="Screen Shot 2020-04-15 at 4 26 50 PM" src="https://user-images.githubusercontent.com/709009/79390811-04942280-7f36-11ea-840e-c234f087100e.png">
9. Don't worry about the fact that nothing is showing in the yellow bar - its because of the 'legacy' setting.
10. Visit https://gradimmuno.local.drupal.uiowa.edu/admin/config/system/site-information and add some parent organization information.
11. Check that https://gradimmuno.local.drupal.uiowa.edu/ is showing the "below" header and has the correct parent name and link in the yellow bar:
    <img width="1350" alt="Screen Shot 2020-04-10 at 6 12 27 PM" src="https://user-images.githubusercontent.com/709009/79028824-477d8100-7b57-11ea-852d-e4b0d9ea121b.png">
12. Visit http://gradimmuno.local.drupal.uiowa.edu/admin/appearance/settings/uids_base and verify that the settings to change the Site name display are disabled.
13. Verify that when you change the Header navigation to something other than 'toggle', the option to turn on Sticky is no longer visible and click save.
14. Visit https://gradimmuno.local.drupal.uiowa.edu/ and use the Inspector to verify that the body has the class 'header-not-sticky' and does not have 'header-sticky'.

# GTM updates

Resolves #1328 

This handles both campus-wide GTM and local site-specific GTM. The campus-wide GTM is loaded as it was previously in uiowa_bar (production only). The local site-specific GTM utilizes the contrib GTM module to allow multiple containers with separate configuration. Though a webmaster can go ahead and configure settings for the container, each container displays a warning notifiying the webmaster that to be inserted and used within the site, they must contact an administrator to enable the functionality. Additionally, non-admins are prevented from configuring the global GTM settings through the contrib module.

For the admins, the local GTM functionality is controlled through a boolean on the uiowa_core settings form. This toggle is checked on GTM insert to decide at the last moment whether or not to insert the configured containers. This uses the contrib module api hook. Additionally, in case GTM does something to the UI interface, the boolean can also be toggled through a custom drush command which will also clear caches and attempt to purge the domain.

Though I believe this resolves the story, deployment must be done in coordination with https://github.com/uiowa/uiowa_bar/pull/55

## To Test

Due to the production assumptions of this module. Testing is kind of complicated. Since this functionality is off by default, I would suggest the following to approve it for PROD where additional testing can occur:

- Temporarily disable the AH_PRODUCTION check in uiowa_core.module locally around line 19 to see whether the campus-wide GTM is loaded similar to the uiowa_bar method.
- Confirm that non-admins can't access:
 - https://sitenowdemo.local.drupal.uiowa.edu/admin/config/system/google-tag/settings
 - https://sitenowdemo.local.drupal.uiowa.edu/admin/config/system/uiowa-core
- Toggle the boolean with the custom drush command: `drush uiowa_core:toggle-gtag`
